banzai_nres/main.py

```diff
@@ -1,40 +1,119 @@
 """
 main.py: Main driver script for the banzai-NRES pipeline.
-    The make_master_bias_console() and make_master_dark_console() are the entry points.
+    reduce_night() is the console entry point.
 Authors
     Curtis McCully (cmccully@lcogt.net)
-July 2018
+
     G. Mirek Brandt (gmbrandt@ucsb.edu)
-July 2018
 """
+import datetime
+
+import banzai_nres.settings as nres_settings  # import to override banzai settings.
+import banzai.settings as banzai_settings
+
+from banzai_nres.utils.db_utils import get_raw_path
+
+from banzai.main import process_directory, parse_directory_args
+from banzai.utils import date_utils
+from banzai import dbs
+from banzai import logs
+from banzai.main import run_master_maker
+
 
-from banzai_nres import settings
-from banzai.main import process_directory, parse_directory_args, run_end_of_night
 import logging
 
 logger = logging.getLogger(__name__)
 
 
-def make_master_bias(pipeline_context=None, raw_path=None):
-    pipeline_context, raw_path = parse_directory_args(pipeline_context, raw_path, settings.NRESSettings())
-    process_directory(pipeline_context, raw_path, ['BIAS'], last_stage=pipeline_context.LAST_STAGE['BIAS'],
-                      extra_stages=pipeline_context.EXTRA_STAGES['BIAS'],
-                      log_message='Making Master BIAS', calibration_maker=True)
+class ReductionCriterion(object):
```

Comment:
I feel like this is just a function. It could check these things on runtime_context. maybe a validate dates and get_frame type or something.

Reply:
added.